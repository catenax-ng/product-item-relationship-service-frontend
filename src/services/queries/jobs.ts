import { useMutation } from "@tanstack/react-query";
import { useQuery } from "../../lib/react-query";
import { IRSRequestBody } from "../../types/jobs";
import { getCurrentEnvironment } from "../../utils/sessionStorageHandling";
import { selectAPI } from "../api/jobs.api";

export const useFetchJobById = ({ id }: { id: string }, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, id], () => jobAPI.fetchJobById(id), {
    refetchInterval,
  });
};

export const useFetchJobs = (page: number, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, page], () => jobAPI.fetchJobs(page), { refetchInterval });
};

export const useCancelJobs = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation(jobAPI.cancelJob);
};

export const useCreateJob = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation({
    mutationFn: (data: IRSRequestBody) => {
      return jobAPI.createJob(data);
    },
  });
};
