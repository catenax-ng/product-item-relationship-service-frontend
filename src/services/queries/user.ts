import { useQuery } from "../../lib";
import { keycloak } from "../../lib/keycloak";

export const getUserInfo = () => {
  return useQuery(["userInfo"], async () => {
    return (await keycloak.loadUserInfo()) as { name: string; tenant: string };
  });
};
