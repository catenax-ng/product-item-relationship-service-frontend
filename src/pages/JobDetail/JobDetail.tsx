import { useParams } from "react-router-dom";
import { useFetchJobById } from "../../services/queries/jobs";
import { ErrorDisplay } from "../ErrorDisplay";
import { IrsJobDetails } from "./features/IrsJobDetails";
import { IRSJobTombstones } from "./features/IRSJobTombstones";
import { IrsJobVisualization } from "./features/IRSJobVisualization/IrsJobVIsualization";

export const JobDetail: React.FC = () => {
  const { jobId = "" } = useParams();
  const { data: job, isError, isLoading, error } = useFetchJobById(jobId);
  if (isLoading) {
    //TODO: Handle Loading State
    console.warn("implement job loading");
    return null;
  }
  if (isError) {
    return <ErrorDisplay error={error} />;
  }

  return (
    <>
      <IrsJobVisualization job={job} />
      <IrsJobDetails job={job.job} />
      <IRSJobTombstones tombstones={job.tombstones} />
    </>
  );
};
