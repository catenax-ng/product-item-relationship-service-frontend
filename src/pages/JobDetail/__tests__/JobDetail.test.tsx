import { JobsDemoDataSuccess } from "../../../services/api/__test__/jobs.exampleData";
import * as queryHooks from "../../../services/queries/jobs";
import { renderWithRouter } from "../../../testing/test-utils";
import { JobDetail } from "../JobDetail";

it("renders a loading state", () => {
  const mockData = {
    data: undefined,
    isError: false,
    isLoading: true,
  };
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  //@ts-ignore
  jest.spyOn(queryHooks, "useFetchJobById").mockReturnValue(mockData);

  const { container } = renderWithRouter(<JobDetail />);
  expect(container).toMatchSnapshot();
});

it("renders a error state", () => {
  const mockData = {
    data: undefined,
    isError: true,
    isLoading: false,
    error: new Error("test error"),
  };
  //TODO: expand with dummy error data
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  //@ts-ignore
  jest.spyOn(queryHooks, "useFetchJobById").mockReturnValue(mockData);

  const { container } = renderWithRouter(<JobDetail />);
  expect(container).toMatchSnapshot();
});

it("renders a data success state", () => {
  const mockData = {
    data: JobsDemoDataSuccess,
    isError: false,
    isLoading: false,
  };
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  //@ts-ignore
  jest.spyOn(queryHooks, "useFetchJobById").mockReturnValue(mockData);

  const { container } = renderWithRouter(<JobDetail />);
  expect(container).toMatchSnapshot();
});
