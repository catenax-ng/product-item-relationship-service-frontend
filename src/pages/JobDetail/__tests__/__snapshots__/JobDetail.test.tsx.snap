// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a data success state 1`] = `
<div>
  <section>
    <div
      class="irs-visualization MuiBox-root css-xi606m"
    >
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="irs-visualization-header MuiBox-root css-0"
        >
          <h5>
            content.irs.visualization.title
          </h5>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeMedium css-1gws2xf-MuiButtonBase-root-MuiIconButton-root"
            style="align-self: right;"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FullscreenIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 14H5v5h5v-2H7v-3zm-2-4h2V7h3V5H5v5zm12 7h-3v2h5v-5h-2v3zM14 5v2h3v3h2V5h-5z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="Canvas-module_container__22cN4 canvas Canvas-module_pannable__1d7xd"
          style="height: 800px; width: 100%;"
        >
          <svg
            height="2000"
            id="ref-1"
            width="2000"
            xmlns="http://www.w3.org/2000/svg"
          >
            <defs>
              <marker
                id="end-arrow"
                markerHeight="8"
                markerWidth="8"
                orient="auto"
                refX="8"
                viewBox="0 -4 8 8"
              >
                <path
                  class="Arrow-module_arrow__aEgOT"
                  d="M0,-4L8,0L0,4"
                  transform="translate(0, 0) rotate(0)"
                />
              </marker>
            </defs>
            <g
              opacity="0"
              style="transform: scale(0); transform-origin: 0px 0px;"
              transform-origin="0px 0px"
            />
          </svg>
        </div>
      </div>
    </div>
  </section>
  <section
    style="padding: 34px 20px;"
  >
    <div
      class="irs-job-details MuiBox-root css-0"
    >
      <div
        class="irs-job-details-header MuiBox-root css-0"
      >
        <h5>
          content.irs.jobDetails.title
        </h5>
      </div>
      <div
        class="irs-job-details-content MuiBox-root css-0"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              Job ID:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              e5347c88-a921-11ec-b909-0242ac120002
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.globalAssetId
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              urn:uuid:6c311d29-5753-46d4-b32c-19b918ea93b0
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              Job State:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              COMPLETED
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.exception
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              <pre
                style="display: block; overflow-x: auto; padding: 0.5em; background: white; color: black;"
              >
                <code
                  style="white-space: pre;"
                >
                  <span>
                    {

                  </span>
                  <span>
                      "
                  </span>
                  <span
                    style="color: rgb(0, 0, 0);"
                  >
                    errorDetail
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ": "
                  </span>
                  <span>
                    Timeout while requesting Digital Registry
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ",

                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                      "
                  </span>
                  <span>
                    exception
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ": "
                  </span>
                  <span>
                    IrsTimeoutException
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ",

                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                      "
                  </span>
                  <span>
                    exceptionDate
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ": "
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    2022
                  </span>
                  <span>
                    -
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    02
                  </span>
                  <span>
                    -
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    03
                  </span>
                  <span>
                    T
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    14
                  </span>
                  <span>
                    :
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    48
                  </span>
                  <span>
                    :
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    54
                  </span>
                  <span>
                    .
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    709
                  </span>
                  <span>
                    Z
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "

                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    }
                  </span>
                </code>
              </pre>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.createdOn
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.startedOn
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.lastModifiedOn
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.jobCompleted
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.owner
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.summary
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              <pre
                style="display: block; overflow-x: auto; padding: 0.5em; background: white; color: black;"
              >
                <code
                  style="white-space: pre;"
                >
                  <span>
                    {

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "asyncFetchedItems"
                  </span>
                  <span>
                    : {

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "completed"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    0
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "failed"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    0
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "running"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    0
                  </span>
                  <span>
                    

                  </span>
                    }

                  }
                </code>
              </pre>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.jobParameter
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              <pre
                style="display: block; overflow-x: auto; padding: 0.5em; background: white; color: black;"
              >
                <code
                  style="white-space: pre;"
                >
                  <span>
                    {

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "aspects"
                  </span>
                  <span>
                    : [

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "SerialPartTypization"
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "ContactInformation"
                  </span>
                  <span>
                    

                  </span>
                    ],

                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "bomLifecycle"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "asBuilt"
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "collectAspects"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    false
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "depth"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    4
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "direction"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "downward"
                  </span>
                  <span>
                    

                  </span>
                  }
                </code>
              </pre>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <section>
    <div
      class="irs-tombstones-details MuiBox-root css-0"
    >
      <div
        class="irs-tombstones-details-header MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-0"
          style="display: inline-block; color: rgb(239, 83, 80); margin-top: 20px;"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ErrorOutlineIcon"
            focusable="false"
            style="font-size: 50px; float: left; vertical-align: middle; margin-top: 10px;"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
          <h2
            style="float: left; margin-left: 10px;"
          >
            content.irs.dialog.submodelTombstones.title
          </h2>
        </div>
      </div>
      <div
        class="irs-tombstones-details-content MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-0"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.dialog.submodelTombstones.lastAttempt:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.dialog.submodelTombstones.errorDetail:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              Details to reason of Failure
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`renders a error state 1`] = `
<div>
  <section>
    <div
      class="irs-error-page MuiBox-root css-0"
    >
      <div
        class="irs-error-page-header MuiBox-root css-0"
      >
        <h1>
          global.errors.title
        </h1>
      </div>
      <div
        class="error-page-body MuiBox-root css-0"
      >
        <p>
          global.errors.description
        </p>
        <p>
          <i>
            test error
          </i>
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`renders a loading state 1`] = `<div />`;
